bentoml/deployment/sagemaker/__init__.py

```diff
@@ -0,0 +1,291 @@
+
+        :param archive_path: Path
+        :param additional_info: Dict
+        :return: String, location to the output snapshot's path
+        """
+        snapshot_path = generate_bentoml_deployment_snapshot_path(self.bento_service.name,
+                                                                  'aws-sagemaker')
+        shutil.copytree(self.archive_path, snapshot_path)
+        with open(os.path.join(snapshot_path, 'nginx.conf'), 'w') as f:
+            f.write(DEFAULT_NGINX_CONFIG)
+        with open(os.path.join(snapshot_path, 'wsgi.py'), 'w') as f:
+            f.write(DEFAULT_WSGI_PY)
+        with open(os.path.join(snapshot_path, 'serve'), 'w') as f:
+            f.write(DEFAULT_SERVE_SCRIPT)
+
+        # We want to give serve '755' permission. Since chmod take octal number, 755 => 493
+        permission = '755'
+        octal_permission = int(permission, 8)
+        os.chmod(os.path.join(snapshot_path, 'serve'), octal_permission)
+
+        execution_role_arn = get_arn_role_from_current_user()
+        ecr_image_path = create_push_image_to_ecr(self.bento_service, snapshot_path)
+        sagemaker_model_info = {
+            "ModelName": self.model_name,
+            "PrimaryContainer": {
+                "ContainerHostname": self.model_name,
+                "Image": ecr_image_path,
+                "Environment": {
+                    "API_NAME": self.api.name
+                }
+            },
+            "ExecutionRoleArn": execution_role_arn,
+        }
+        logger.info('Creating sagemaker model %s', self.model_name)
+        create_model_response = self.sagemaker_client.create_model(**sagemaker_model_info)
+        logger.debug(create_model_response)
+
+        production_variants = [{
+            "VariantName": self.bento_service.name,
+            "ModelName": self.model_name,
+            "InitialInstanceCount": self.instance_count,
+            "InstanceType": self.instant_type,
+        }]
+        logger.info('Creating sagemaker endpoint %s configuration', self.endpoint_config_name)
+        create_endpoint_config_response = self.sagemaker_client.create_endpoint_config(
+            EndpointConfigName=self.endpoint_config_name, ProductionVariants=production_variants)
+        logger.debug(create_endpoint_config_response)
+        logger.info('Creating sagemaker endpoint %s', self.bento_service.name)
+        create_endpoint_response = self.sagemaker_client.create_endpoint(
+            EndpointName=self.bento_service.name, EndpointConfigName=self.endpoint_config_name)
+        logger.debug(create_endpoint_response)
+
+        # TODO: maybe wait for this endpoint from creating to running and then return
+        endpoint_status = self.sagemaker_client.describe_endpoint(
+            EndpointName=self.bento_service.name)
+        logger.debug(endpoint_status)
+        logger.info(endpoint_status)
+        return snapshot_path
+
+    def check_status(self, display_cli_message=True):
```

Comment:
another way to do this `display_cli_message=True` is to have `check_status` return both status boolean and a status msg: `status, status_message = deployment.check_status()`

and the callee will decide whether or not to print that message to end user

Reply:
since the caller is already decide whether or not print the message to user, i will keep this format for now.

Reply:
done